### ISSUE TITLE
Delisting return adjustment

### ISSUE THREAD

**Issue by mk0417** [OpenSourceAP/CrossSection#49]
Line 43 in **Signals/Code/DataDownloads/I_CRSPmonthly.do**: `replace ret = ret + dlret`.
In this way, 62 observations of delisting adujsted ret are less than -1. The minimum ret is -1.99

Is it better to use `replace ret = (1+ret) * (1+dlret) - 1` as suggested by WRDS? There will be no delisting adjusted return that is less than -1.

**Comment by chenandrewy** [comment]
Yes, I agree @mk0417 .  Thanks! Here's a link to the WRDS suggestion:

https://wrds-www.wharton.upenn.edu/pages/support/support-articles/crsp/stock/using-ret-return-variable-and-dlret-delisting-return-determine-return-last-trading-date/

and I think we can add this to the to do list in the update for early 2022.

**Comment by chenandrewy** [comment]
Delisting returns are used twice in the repo: (1) in Signals/ and (2) in Portfolios/.  (2) is probably much more important, and we should probably close this issue out after updating (2).

**Comment by chenandrewy** [comment]
I think I disagree with the emailed suggestion 1 (on (dlstcd == 500 | (dlstcd >=520 & dlstcd <=584))).  Shumway's (1997) table I looks like his adjustments matter for all dlistcd 500 or anything in 520-584: 

![image](https://user-images.githubusercontent.com/65506481/155228061-892ede01-a2c7-449b-af17-3761b0483ab6.png)

The footnote at the bottom has the list the emailed suggestion points out (500, 520, 574, 580, 584 and (551 to 584)), but my read is that this list is for illustrative purposes only.  I think the other delisting codes are omitted just because they're hard to describe in a short footnote.  Here are some descriptions from crsp.org:

575	Delisted by current exchange - company request, offer rescinded, issue withdrawn by underwriter.

581	Delisted by current exchange - failure to register under 12G of Securities Exchange Act.

I also note that Shumway states: 
![image](https://user-images.githubusercontent.com/65506481/155228512-7bf2b296-df4e-4f75-8f10-12d166e6a006.png)
so 581 should be a performance related problem.  

Long story short: I'm not an expert on delistings, but the code is fine with regards to emailed suggestion 1.

**Comment by chenandrewy** [comment]
Regarding emailed suggestion 2: 

Q1: yes, we should be using the compounding version (1+ret)(1+dlret)-1.  

Q2: this is a tricky issue and my view is that we should not move the return further out.  Because, on the performance-related delisting event, clearly the market value has dropped, but we don't have a market value (it's delisted). So even if the dlret payment doesn't come for a couple months, it's still the best proxy we have for the unobserved market value.

**Comment by chenandrewy** [comment]
@mk0417's comment is finally fixed for Signals/Code/DataDownloads/I_CRSPmonthly.do here: 1b25a909929da27056f6538f47971d3fc861bc5e.

Still got to patch up the portfolios code

**Comment by chenandrewy** [comment]
Patched Portforlios/Code/11_ProcessCRSP.R 5e08b088647b076295f9883aa467be9428c8c4b9
